tests/run_tests.sh

```diff
@@ -26,22 +61,24 @@ for PYTHON in /opt/python/*/bin/python; do
 	$PYTHON $MY_DIR/ssl-check.py
 	IMPLEMENTATION=$(${PYTHON} -c "import sys; print(sys.implementation.name)")
 	PYVERS=$(${PYTHON} -c "import sys; print('.'.join(map(str, sys.version_info[:2])))")
-	if [ "${IMPLEMENTATION}" == "pypy" ]; then
-		LINK_PREFIX=pypy
-	else
-		LINK_PREFIX=python
+	if [ "${IMPLEMENTATION}" == "cpython" ]; then
 		# Make sure sqlite3 module can be loaded properly and is the manylinux version one
 		# c.f. https://github.com/pypa/manylinux/issues/1030
 		$PYTHON -c 'import sqlite3; print(sqlite3.sqlite_version); assert sqlite3.sqlite_version_info[0:2] >= (3, 34)'
 		# Make sure tkinter module can be loaded properly
 		$PYTHON -c 'import tkinter; print(tkinter.TkVersion); assert tkinter.TkVersion >= 8.6'
+		# cpython shall be available as python
+		LINK_VERSION=$(python${PYVERS} -VV)
+		REAL_VERSION=$(${PYTHON} -VV)
+		test "${LINK_VERSION}" = "${REAL_VERSION}"
 	fi
-	# pythonX.Y / pypyX.Y shall be available directly in PATH
-	LINK_VERSION=$(${LINK_PREFIX}${PYVERS} -V)
-	REAL_VERSION=$(${PYTHON} -V)
+	# cpythonX.Y / pypyX.Y shall be available directly in PATH
+	LINK_VERSION=$(${IMPLEMENTATION}${PYVERS} -VV)
+	REAL_VERSION=$(${PYTHON} -VV)
 	test "${LINK_VERSION}" = "${REAL_VERSION}"
 
 	# check a simple project can be built
+	if [ "${IMPLEMENTATION}" != "graalpy" ]; then
```

Comment:
comment back ported from #1518 

mayeut wrote:
> It seems that packaging does not agree with wheel w.r.t. to the ABI tag. This is the sign that something bad will happen in the future (I guess when pip will update packaging: https://github.com/pypa/pip/issues/11715).
The difference comes from https://github.com/pypa/packaging/commit/30554f503a5368aabf6ee571bd9cdf29d5f00c12
>
> Unfortunately, the ecosystem did not move forward with a sysconfig like that to express this ABI. xref https://discuss.python.org/t/sysconfig-should-provide-more-information-to-help-packaging/22950 and https://github.com/pypa/packaging/issues/330

timfel wrote:
> We have a PR to wheel to fix the get_abi_tag function in src/wheel/bdist_wheel.py. When I signed off on the change in packaging, I actually expected pip to upgrade their vendored version quicker. Right now GraalPy is in this in-between state there, with pip having the old vendored packaging, wheel having a copy the logic in [bdist_wheel.py](https://github.com/pypa/wheel/blob/main/src/wheel/bdist_wheel.py#L91), and packages like maturin and meson-python already implementing the new logic. I hope it'll just slowly resolve as the necessary changes that started with that packaging commit slowly perculate through everywhere.

mayeut wrote:
> xref https://github.com/python/cpython/issues/99560

timfel wrote:
> Here's the wheel PR: https://github.com/pypa/wheel/pull/557

mayeut wrote:
> There's something wrong with platform tag, I guess this is related to the is_32bit check.
xref https://github.com/pypa/packaging/pull/614

timfel wrote:
> Yes, the is_32bit check is biting us. The PR you linked didn't move forward mostly because of concern for 32-bit macOS. We have found making sys.maxsize larger on GraalPy brings other issues though, since some code assumes that (as documented) this gives the maximum array size.
I took this opportunity to https://github.com/pypa/packaging/pull/614#issuecomment-1679459182 on the closed PR to packaging when we might be able to fix this.